packages/grid_client/src/helpers/validator.ts

```diff
@@ -31,4 +31,106 @@ function validateHexSeed(seed: string, length: number): boolean {
   return true;
 }
 
-export { validateObject, validateInput, validateHexSeed };
+interface ValidationOptions {
+  props?: boolean | string | string[];
+  methods?: boolean | string | string[];
+}
+
+function Validate(options?: ValidationOptions): ClassDecorator {
```

Comment:
Can you please provide docstrings for all of the created functions and describe the purpose of each function? Also, we typically use the `validateInput` as a decorator in the grid client. I know it's different, but at the very least we need to change the name of the newly created decorator. what do you think?